--- a/setup.py
+++ b/setup.py
@@ -6,6 +6,7 @@
 import sys, os, imp, re, optparse
 from glob import glob
 from platform import machine as platform_machine
+import subprocess
 import sysconfig
 
 from distutils import log
@@ -431,6 +432,8 @@
             '/lib', '/usr/lib',
             ]
         inc_dirs = self.compiler.include_dirs + ['/usr/include']
+        gnu_triplet = os.popen('dpkg-architecture -qDEB_HOST_GNU_TYPE').readline()[:-1]; print 'XXX', gnu_triplet
+        inc_dirs.append(os.path.join('/usr/include', gnu_triplet))
         exts = []
         missing = []
 
