--- a/configure
+++ b/configure
@@ -12206,39 +12206,28 @@
 if ${ac_cv_posix_semaphores_enabled+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then :
   ac_cv_posix_semaphores_enabled=yes
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <unistd.h>
-#include <fcntl.h>
-#include <stdio.h>
-#include <semaphore.h>
-#include <sys/stat.h>
-
-int main(void) {
-  sem_t *a = sem_open("/autoconf", O_CREAT, S_IRUSR|S_IWUSR, 0);
-  if (a == SEM_FAILED) {
-    perror("sem_open");
-    return 1;
-  }
-  sem_close(a);
-  sem_unlink("/autoconf");
-  return 0;
-}
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  ac_cv_posix_semaphores_enabled=yes
-else
-  ac_cv_posix_semaphores_enabled=no
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
+#AC_RUN_IFELSE([AC_LANG_SOURCE([[
+##include <unistd.h>
+##include <fcntl.h>
+##include <stdio.h>
+##include <semaphore.h>
+##include <sys/stat.h>
+
+#int main(void) {
+#  sem_t *a = sem_open("/autoconf", O_CREAT, S_IRUSR|S_IWUSR, 0);
+#  if (a == SEM_FAILED) {
+#    perror("sem_open");
+#    return 1;
+#  }
+#  sem_close(a);
+#  sem_unlink("/autoconf");
+#  return 0;
+#}
+#]])],
+#[ac_cv_posix_semaphores_enabled=yes],
+#[ac_cv_posix_semaphores_enabled=no],
+#[ac_cv_posix_semaphores_enabled=yes])
 
 fi
 
@@ -12257,43 +12246,32 @@
 if ${ac_cv_broken_sem_getvalue+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  if test "$cross_compiling" = yes; then :
-  ac_cv_broken_sem_getvalue=yes
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <unistd.h>
-#include <fcntl.h>
-#include <stdio.h>
-#include <semaphore.h>
-#include <sys/stat.h>
-
-int main(void){
-  sem_t *a = sem_open("/autocftw", O_CREAT, S_IRUSR|S_IWUSR, 0);
-  int count;
-  int res;
-  if(a==SEM_FAILED){
-    perror("sem_open");
-    return 1;
-
-  }
-  res = sem_getvalue(a, &count);
-  sem_close(a);
-  sem_unlink("/autocftw");
-  return res==-1 ? 1 : 0;
-}
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
   ac_cv_broken_sem_getvalue=no
-else
-  ac_cv_broken_sem_getvalue=yes
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
+#AC_RUN_IFELSE([AC_LANG_SOURCE([[
+##include <unistd.h>
+##include <fcntl.h>
+##include <stdio.h>
+##include <semaphore.h>
+##include <sys/stat.h>
+
+#int main(void){
+#  sem_t *a = sem_open("/autocftw", O_CREAT, S_IRUSR|S_IWUSR, 0);
+#  int count;
+#  int res;
+#  if(a==SEM_FAILED){
+#    perror("sem_open");
+#    return 1;
+
+#  }
+#  res = sem_getvalue(a, &count);
+#  sem_close(a);
+#  sem_unlink("/autocftw");
+#  return res==-1 ? 1 : 0;
+#}
+#]])],
+#[ac_cv_broken_sem_getvalue=no],
+#[ac_cv_broken_sem_getvalue=yes],
+#[ac_cv_broken_sem_getvalue=yes])
 
 fi
 
--- a/configure.in
+++ b/configure.in
@@ -3609,28 +3609,28 @@
 # isn't loaded by default, so an attempt to call
 # sem_open results in a 'Signal 12' error.
 AC_MSG_CHECKING(whether POSIX semaphores are enabled)
-AC_CACHE_VAL(ac_cv_posix_semaphores_enabled,
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <unistd.h>
-#include <fcntl.h>
-#include <stdio.h>
-#include <semaphore.h>
-#include <sys/stat.h>
+AC_CACHE_VAL(ac_cv_posix_semaphores_enabled,ac_cv_posix_semaphores_enabled=yes
+#AC_RUN_IFELSE([AC_LANG_SOURCE([[
+##include <unistd.h>
+##include <fcntl.h>
+##include <stdio.h>
+##include <semaphore.h>
+##include <sys/stat.h>
 
-int main(void) {
-  sem_t *a = sem_open("/autoconf", O_CREAT, S_IRUSR|S_IWUSR, 0);
-  if (a == SEM_FAILED) {
-    perror("sem_open");
-    return 1;
-  }
-  sem_close(a);
-  sem_unlink("/autoconf");
-  return 0;
-}
-]])],
-[ac_cv_posix_semaphores_enabled=yes],
-[ac_cv_posix_semaphores_enabled=no],
-[ac_cv_posix_semaphores_enabled=yes])
+#int main(void) {
+#  sem_t *a = sem_open("/autoconf", O_CREAT, S_IRUSR|S_IWUSR, 0);
+#  if (a == SEM_FAILED) {
+#    perror("sem_open");
+#    return 1;
+#  }
+#  sem_close(a);
+#  sem_unlink("/autoconf");
+#  return 0;
+#}
+#]])],
+#[ac_cv_posix_semaphores_enabled=yes],
+#[ac_cv_posix_semaphores_enabled=no],
+#[ac_cv_posix_semaphores_enabled=yes])
 )
 AC_MSG_RESULT($ac_cv_posix_semaphores_enabled)
 if test $ac_cv_posix_semaphores_enabled = no
@@ -3641,32 +3641,32 @@
 
 # Multiprocessing check for broken sem_getvalue
 AC_MSG_CHECKING(for broken sem_getvalue)
-AC_CACHE_VAL(ac_cv_broken_sem_getvalue,
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <unistd.h>
-#include <fcntl.h>
-#include <stdio.h>
-#include <semaphore.h>
-#include <sys/stat.h>
+AC_CACHE_VAL(ac_cv_broken_sem_getvalue,ac_cv_broken_sem_getvalue=no
+#AC_RUN_IFELSE([AC_LANG_SOURCE([[
+##include <unistd.h>
+##include <fcntl.h>
+##include <stdio.h>
+##include <semaphore.h>
+##include <sys/stat.h>
 
-int main(void){
-  sem_t *a = sem_open("/autocftw", O_CREAT, S_IRUSR|S_IWUSR, 0);
-  int count;
-  int res;
-  if(a==SEM_FAILED){
-    perror("sem_open");
-    return 1;
+#int main(void){
+#  sem_t *a = sem_open("/autocftw", O_CREAT, S_IRUSR|S_IWUSR, 0);
+#  int count;
+#  int res;
+#  if(a==SEM_FAILED){
+#    perror("sem_open");
+#    return 1;
 
-  }
-  res = sem_getvalue(a, &count);
-  sem_close(a);
-  sem_unlink("/autocftw");
-  return res==-1 ? 1 : 0;
-}
-]])],
-[ac_cv_broken_sem_getvalue=no],
-[ac_cv_broken_sem_getvalue=yes],
-[ac_cv_broken_sem_getvalue=yes])
+#  }
+#  res = sem_getvalue(a, &count);
+#  sem_close(a);
+#  sem_unlink("/autocftw");
+#  return res==-1 ? 1 : 0;
+#}
+#]])],
+#[ac_cv_broken_sem_getvalue=no],
+#[ac_cv_broken_sem_getvalue=yes],
+#[ac_cv_broken_sem_getvalue=yes])
 )
 AC_MSG_RESULT($ac_cv_broken_sem_getvalue)
 if test $ac_cv_broken_sem_getvalue = yes
