--- a/configure.in
+++ b/configure.in
@@ -3623,28 +3623,28 @@
 # isn't loaded by default, so an attempt to call
 # sem_open results in a 'Signal 12' error.
 AC_MSG_CHECKING(whether POSIX semaphores are enabled)
-AC_CACHE_VAL(ac_cv_posix_semaphores_enabled,
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <unistd.h>
-#include <fcntl.h>
-#include <stdio.h>
-#include <semaphore.h>
-#include <sys/stat.h>
+AC_CACHE_VAL(ac_cv_posix_semaphores_enabled,ac_cv_posix_semaphores_enabled=yes
+#AC_RUN_IFELSE([AC_LANG_SOURCE([[
+##include <unistd.h>
+##include <fcntl.h>
+##include <stdio.h>
+##include <semaphore.h>
+##include <sys/stat.h>
 
-int main(void) {
-  sem_t *a = sem_open("/autoconf", O_CREAT, S_IRUSR|S_IWUSR, 0);
-  if (a == SEM_FAILED) {
-    perror("sem_open");
-    return 1;
-  }
-  sem_close(a);
-  sem_unlink("/autoconf");
-  return 0;
-}
-]])],
-[ac_cv_posix_semaphores_enabled=yes],
-[ac_cv_posix_semaphores_enabled=no],
-[ac_cv_posix_semaphores_enabled=yes])
+#int main(void) {
+#  sem_t *a = sem_open("/autoconf", O_CREAT, S_IRUSR|S_IWUSR, 0);
+#  if (a == SEM_FAILED) {
+#    perror("sem_open");
+#    return 1;
+#  }
+#  sem_close(a);
+#  sem_unlink("/autoconf");
+#  return 0;
+#}
+#]])],
+#[ac_cv_posix_semaphores_enabled=yes],
+#[ac_cv_posix_semaphores_enabled=no],
+#[ac_cv_posix_semaphores_enabled=yes])
 )
 AC_MSG_RESULT($ac_cv_posix_semaphores_enabled)
 if test $ac_cv_posix_semaphores_enabled = no
@@ -3655,32 +3655,32 @@
 
 # Multiprocessing check for broken sem_getvalue
 AC_MSG_CHECKING(for broken sem_getvalue)
-AC_CACHE_VAL(ac_cv_broken_sem_getvalue,
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <unistd.h>
-#include <fcntl.h>
-#include <stdio.h>
-#include <semaphore.h>
-#include <sys/stat.h>
+AC_CACHE_VAL(ac_cv_broken_sem_getvalue,ac_cv_broken_sem_getvalue=no
+#AC_RUN_IFELSE([AC_LANG_SOURCE([[
+##include <unistd.h>
+##include <fcntl.h>
+##include <stdio.h>
+##include <semaphore.h>
+##include <sys/stat.h>
 
-int main(void){
-  sem_t *a = sem_open("/autocftw", O_CREAT, S_IRUSR|S_IWUSR, 0);
-  int count;
-  int res;
-  if(a==SEM_FAILED){
-    perror("sem_open");
-    return 1;
+#int main(void){
+#  sem_t *a = sem_open("/autocftw", O_CREAT, S_IRUSR|S_IWUSR, 0);
+#  int count;
+#  int res;
+#  if(a==SEM_FAILED){
+#    perror("sem_open");
+#    return 1;
 
-  }
-  res = sem_getvalue(a, &count);
-  sem_close(a);
-  sem_unlink("/autocftw");
-  return res==-1 ? 1 : 0;
-}
-]])],
-[ac_cv_broken_sem_getvalue=no],
-[ac_cv_broken_sem_getvalue=yes],
-[ac_cv_broken_sem_getvalue=yes])
+#  }
+#  res = sem_getvalue(a, &count);
+#  sem_close(a);
+#  sem_unlink("/autocftw");
+#  return res==-1 ? 1 : 0;
+#}
+#]])],
+#[ac_cv_broken_sem_getvalue=no],
+#[ac_cv_broken_sem_getvalue=yes],
+#[ac_cv_broken_sem_getvalue=yes])
 )
 AC_MSG_RESULT($ac_cv_broken_sem_getvalue)
 if test $ac_cv_broken_sem_getvalue = yes
