--- a/Lib/pydoc.py
+++ b/Lib/pydoc.py
@@ -1964,10 +1964,14 @@
         if modname[-9:] == '.__init__':
             modname = modname[:-9] + ' (package)'
         print modname, desc and '- ' + desc
+    def onerror(modname):
+        # Ignore non-ImportError exceptions raised whilst trying to
+        # import modules
+        pass
     try: import warnings
     except ImportError: pass
     else: warnings.filterwarnings('ignore') # ignore problems during import
-    ModuleScanner().run(callback, key)
+    ModuleScanner().run(callback, key, onerror=onerror)
 
 # --------------------------------------------------- web browser interface
 
