# DP: Unexpand revision numbers

--- a/configure.in
+++ b/configure.in
@@ -14,7 +14,7 @@
 ])
 version_required(2.65)
 
-AC_REVISION($Revision: 86076 $)
+AC_REVISION($Revision$)
 AC_INIT(python, PYTHON_VERSION, http://bugs.python.org/)
 AC_CONFIG_SRCDIR([Include/object.h])
 AC_CONFIG_HEADER(pyconfig.h)
--- a/Include/patchlevel.h
+++ b/Include/patchlevel.h
@@ -31,7 +31,7 @@
 /*--end constants--*/
 
 /* Subversion Revision number of this file (not of the repository) */
-#define PY_PATCHLEVEL_REVISION  "$Revision: 86832 $"
+#define PY_PATCHLEVEL_REVISION  "$Revision$"
 
 /* Version as a single 4-byte hex number, e.g. 0x010502B2 == 1.5.2b2.
    Use this for numeric comparisons, e.g. #if PY_VERSION_HEX >= ... */
--- a/Python/sysmodule.c
+++ b/Python/sysmodule.c
@@ -1144,7 +1144,7 @@
 
 /* Subversion branch and revision management */
 static const char _patchlevel_revision[] = PY_PATCHLEVEL_REVISION;
-static const char headurl[] = "$HeadURL: http://svn.python.org/projects/python/tags/r271/Python/sysmodule.c $";
+static const char headurl[] = "$HeadURL$";
 static int svn_initialized;
 static char patchlevel_revision[50]; /* Just the number */
 static char branch[50];
